virt-v2v/cold/scripts/rhel/run/network_config_util.sh

```diff
@@ -1,145 +1,92 @@
-            interface_names_rhel9+=("$interface_name")
-        else
-            interface_names_rhel9+=("")
+            echo "SUBSYSTEM==\"net\",ACTION==\"add\",ATTR{address}==\"$S_HW\",NAME=\"$DEVICE\""
         fi
-    done
-
-    # Take the interface name from rhel8 config files, if missing fallback to rhel9
-    for i in "${!ip_addresses[@]}"; do
-        interface_name="${interface_names_rhel8[$i]:-${interface_names_rhel9[$i]}}"
-        interface_names+=("${interface_name}")
-    done
-
-    log "Found interfaces: ${interface_names[@]}"
-
-    echo "${interface_names[@]}"
+    done < "$V2V_MAP_FILE"
 }
 
-# Main function to run the script
-main() {
-    # /tmp/macToIP is auto generated by RHEL customization code,
-    # it contains the value of os.Getenv("V2V_staticIPs") and is added to the
-    # guest filesystem using go code.
-    # Ref: package:main method:handleStaticIPConfiguration
-    local input_file="${1:-/tmp/macToIP}"
+# Checks for duplicate hardware addresses 
+check_dupe_hws() {
+    input=$(cat)
 
-    # Default input file path and network scripts and connections directory
-    local network_scripts_dir="${2:-/etc/sysconfig/network-scripts/}"
-    local network_connections_dir="${3:-/etc/NetworkManager/system-connections/}"
-    
-    # Udev rules file argument or default location
-    local udev_rules_file="${4:-/etc/udev/rules.d/70-persistent-net.rules}"
-    
-    # Check if udev rules file exists
-    if [ -f "$udev_rules_file" ]; then
-        echo "File $udev_rules_file already exists. Exiting."
-        exit 0
-    fi
-
-    # Parse MAC and IP addresses and create the mapping
-    local mac_addresses=($(parse_mac_to_ip "$input_file" | head -n 1))
-    local ip_addresses=($(parse_mac_to_ip "$input_file" | tail -n 1))
-    local interface_names=($(create_ip_mac_interface_mapping "$input_file" "$network_scripts_dir" "$network_connections_dir"))
-
-    log "Building ${udev_rules_file}"
+    # Extract MAC addresses, convert to uppercase, sort them, and find duplicates
+    dupes=$(grep -io -E "[0-9A-F:]{17}" <<< "$input" | tr 'a-f' 'A-F' | sort | uniq -d)
 
-    # Add entries to the udev rules file
-    for i in "${!mac_addresses[@]}"; do
-        local mac="${mac_addresses[$i]}"
-        local interface="${interface_names[$i]}"
+    # If duplicates are found, print an error and exit
+    if [ -n "$dupes" ]; then
+        echo "ERR: Duplicate hw"
+        exit 2
+    fi
 
-        if [ -n "$mac" ] && [ -n "$interface" ]; then
-            echo "SUBSYSTEM==\"net\",ACTION==\"add\",ATTR{address}==\"$mac\",NAME=\"$interface\"" >> "$udev_rules_file"
-            echo "Added entry: MAC=$mac, Interface=$interface to $udev_rules_file"
-        fi
-    done
+    echo "$input"
 }
 
-# Call the main function with arguments if provided
-main "$@"
\ No newline at end of file
+# Create udev rules check for duplicates and write them to udev file
+{
+    udev_from_ifcfg
+    udev_from_nm
+} | check_dupe_hws > "$UDEV_RULES_FILE"
```

Comment:
Can we keep this in main()

thhen it's easier to reuse in tests

Reply:
also changed in the test to use it :+1: